map/lanelet2_extension/README.md

```diff
@@ -0,0 +1,87 @@
+# lanelet2_extension package
+
+This package contains external library for Lanelet2 and is meant to ease the use of Lanelet2 in Autoware.
+
+## Lanelet Format for Autoware
+
+Autoware uses extended Lanelet2 Format for Autoware, which means you need to add some tags to default OSM file if you want to fully use Lanelet2 maps. For details about custom tags, please refer to this [document](./docs/lanelet2_format_extension.md).
+
+## Code API
+
+### IO
+
+#### Autoware OSM Parser
+
+Autoware Lanelet2 Format uses .osm extension as original Lanelet2.
+However, there are some custom tags that is used by the parser.
+
+Currently, this includes:
+
+- overwriting x,y values with `local_x` and `local_y` tags.
+- reading `<MapMetaInfo>` tag which contains information about map format version and map version.
+
+The parser is registered as "autoware_osm_handler" as lanelet parser
+
+### Projection
+
+#### MGRS Projector
+
+MGRS projector projects latitude longitude into MGRS Coordinates.
+
+### Regulatory Elements
+
+#### Autoware Traffic Light
+
+Autoware Traffic Light class allows you to retrieve information about traffic lights.
+Autoware Traffic Light class contains following members:
+
+- traffic light shape
+- light bulbs information of traffic lights
+- stopline associated to traffic light
+
+### Utility
+
+#### Message Conversion
+
+This contains functions to convert lanelet map objects into ROS messages.
+Currently it contains following conversions:
+
+- lanelet::LaneletMapPtr to/from lanelet_msgs::MapBinMsg
+- lanelet::Point3d to geometry_msgs::Point
+- lanelet::Point2d to geometry_msgs::Point
+- lanelet::BasicPoint3d to geometry_msgs::Point
```

Comment:
This description is old.

Reply:
87cba84
lanelet_msgs::MapBinMsg -> autoware_auto_mapping_msgs::msg::HADMapBin
and others are same message type